--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -524,8 +524,8 @@
   option(GDCM_USE_JPEGLS "Build GDCM with JPEG-LS support" ON)
   mark_as_advanced(GDCM_USE_JPEGLS)
   if(GDCM_WRAP_JAVA)
-    set(GDCM_JAVA_SOURCE_VERSION "1.7" CACHE STRING "javac source version")
-    set(GDCM_JAVA_TARGET_VERSION "1.7" CACHE STRING "javac target version")
+    set(GDCM_JAVA_SOURCE_VERSION "1.5" CACHE STRING "javac source version")
+    set(GDCM_JAVA_TARGET_VERSION "1.5" CACHE STRING "javac target version")
     mark_as_advanced(GDCM_JAVA_SOURCE_VERSION GDCM_JAVA_TARGET_VERSION)
   endif()
   if(GDCM_WRAP_CSHARP)
--- a/Wrapping/Java/CMakeLists.txt
+++ b/Wrapping/Java/CMakeLists.txt
@@ -14,7 +14,7 @@
 
 # $ export JAVA_HOME=/usr/lib/j2sdk1.6-sun/
 # $ export JAVA_HOME=/usr/lib/jvm/java-1.5.0-sun/
-find_package(Java 1.7 REQUIRED) # javac, jar
+find_package(Java 1.5 REQUIRED) # javac, jar
 find_package(JNI REQUIRED)
 include_directories(
   ${JNI_INCLUDE_DIRS}
