From 5083513138fc90aa602b93293d2e44ae73e883b0 Mon Sep 17 00:00:00 2001
From: Mathieu Malaterre <mathieu.malaterre@gmail.com>
Date: Thu, 18 Oct 2018 10:33:48 +0200
Subject: [PATCH] Adapt to new poppler API

Bug #462
---
 Applications/Cxx/gdcminfo.cxx | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/Applications/Cxx/gdcminfo.cxx b/Applications/Cxx/gdcminfo.cxx
index d926a2a8e..2c510d7df 100644
--- a/Applications/Cxx/gdcminfo.cxx
+++ b/Applications/Cxx/gdcminfo.cxx
@@ -189,7 +189,7 @@ static int checkdeflated(const char *name)
 static std::string getInfoDate(Dict *infoDict, const char *key)
 {
   Object obj;
-  char *s;
+  const char *s;
   int year, mon, day, hour, min, sec, n;
   struct tm tmStruct;
   //char buf[256];
@@ -201,7 +201,8 @@ static std::string getInfoDate(Dict *infoDict, const char *key)
   if (infoDict->lookup((char*)key, &obj)->isString())
 #endif
     {
-    s = obj.getString()->getCString();
+    const GooString* gs = obj.getString();
+    s = gs->getCString();
     if (s[0] == 'D' && s[1] == ':')
       {
       s += 2;
@@ -256,7 +257,7 @@ static std::string getInfoDate(Dict *infoDict, const char *key)
 static std::string getInfoString(Dict *infoDict, const char *key, UnicodeMap *uMap)
 {
   Object obj;
-  GooString *s1;
+  const GooString *s1;
   GBool isUnicode;
   Unicode u;
   char buf[8];
Index: gdcm-2.8.7/Applications/Cxx/gdcmpdf.cxx
===================================================================
--- gdcm-2.8.7.orig/Applications/Cxx/gdcmpdf.cxx
+++ gdcm-2.8.7/Applications/Cxx/gdcmpdf.cxx
@@ -37,7 +37,7 @@
 static std::string getInfoDate(Dict *infoDict, const char *key)
 {
   Object obj;
-  char *s;
+  const char *s;
   int year, mon, day, hour, min, sec, n;
   struct tm tmStruct;
   //char buf[256];
@@ -104,7 +104,7 @@
 static std::string getInfoString(Dict *infoDict, const char *key, UnicodeMap *uMap, GBool & unicode)
 {
   Object obj;
-  GooString *s1;
+  const GooString *s1;
   GBool isUnicode = gFalse;
   Unicode u;
   char buf[8];
