Description: Fix compiling with CharLS 2.0
Author: Gert Wollny <gewo@debian.org>

--- a/Source/MediaStorageAndFileFormat/gdcmJPEGLSCodec.cxx
+++ b/Source/MediaStorageAndFileFormat/gdcmJPEGLSCodec.cxx
@@ -22,6 +22,12 @@
 // CharLS includes
 #include "gdcm_charls.h"
 
+#include <cstring>
+
+using BYTE=unsigned char;
+#define ILV_LINE charls::InterleaveMode::Line;
+#define ILV_NONE charls::InterleaveMode::None;
+
 #if defined(__GNUC__) && GCC_VERSION < 50101
 #pragma GCC diagnostic ignored "-Wmissing-field-initializers"
 #endif
@@ -287,7 +293,7 @@
   provide the possiblity to tune the JPEG-LS internals for better compression
   ratios. Expect a lot of work and testing to achieve small improvements.
 
-  Lossy/lossless is controlled by the field allowedlossyerror. If you put in
+  Lossy/lossless is controlled by the field allowedLossyError. If you put in
   0, encoding is lossless. If it is non-zero, then encoding is lossy. The
   value of 3 is often suggested as a default.
 
--- a/Utilities/gdcm_charls.h
+++ b/Utilities/gdcm_charls.h
@@ -20,11 +20,6 @@
 // It is expected that version 2.0.0 is used
 # include <CharLS/charls.h>
 # include <CharLS/publictypes.h>
-# include <CharLS/util.h>
-# include <CharLS/defaulttraits.h>
-# include <CharLS/losslesstraits.h>
-# include <CharLS/colortransform.h>
-# include <CharLS/processline.h>
 #else
 #include "gdcmcharls/charls.h"
 #include "gdcmcharls/publictypes.h"
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -367,7 +367,7 @@
 
 if(GDCM_USE_SYSTEM_CHARLS)
   find_package(CharLS 2.0.0 REQUIRED)
-  set(GDCM_CHARLS_LIBRARIES ${CHARLS_LIBRARIES})
+  set(GDCM_CHARLS_LIBRARIES ${CHARLS_LIBRARY})
 else()
   set(GDCM_CHARLS_LIBRARIES gdcmcharls)
 endif()
