Description: Fix build with poppler 0.71
Author: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
Forwarded: https://github.com/malaterre/GDCM/pull/70
Last-Update: 2018-11-28

--- a/Applications/Cxx/gdcminfo.cxx
+++ b/Applications/Cxx/gdcminfo.cxx
@@ -258,7 +258,7 @@
 {
   Object obj;
   const GooString *s1;
-  GBool isUnicode;
+  bool isUnicode;
   Unicode u;
   char buf[8];
   int i, n;
@@ -274,12 +274,12 @@
     if ((s1->getChar(0) & 0xff) == 0xfe &&
       (s1->getChar(1) & 0xff) == 0xff)
       {
-      isUnicode = gTrue;
+      isUnicode = true;
       i = 2;
       }
     else
       {
-      isUnicode = gFalse;
+      isUnicode = false;
       i = 0;
       }
     while (i < obj.getString()->getLength())
@@ -539,10 +539,10 @@
       int pages = doc->getNumPages();
       const char *encrypted = doc->isEncrypted() ? "yes" : "no";
       //  printf("yes (print:%s copy:%s change:%s addNotes:%s)\n",
-      //   doc->okToPrint(gTrue) ? "yes" : "no",
-      //   doc->okToCopy(gTrue) ? "yes" : "no",
-      //   doc->okToChange(gTrue) ? "yes" : "no",
-      //   doc->okToAddNotes(gTrue) ? "yes" : "no");
+      //   doc->okToPrint(true) ? "yes" : "no",
+      //   doc->okToCopy(true) ? "yes" : "no",
+      //   doc->okToChange(true) ? "yes" : "no",
+      //   doc->okToAddNotes(true) ? "yes" : "no");
 
       // print linearization info
       const char *optimized = doc->isLinearized() ? "yes" : "no";
--- a/Applications/Cxx/gdcmpdf.cxx
+++ b/Applications/Cxx/gdcmpdf.cxx
@@ -102,11 +102,11 @@
   return out;
 }
 
-static std::string getInfoString(Dict *infoDict, const char *key, UnicodeMap *uMap, GBool & unicode)
+static std::string getInfoString(Dict *infoDict, const char *key, UnicodeMap *uMap, bool & unicode)
 {
   Object obj;
   const GooString *s1;
-  GBool isUnicode = gFalse;
+  bool isUnicode = false;
   Unicode u;
   char buf[8];
   int i, n;
@@ -122,12 +122,12 @@
     if ((s1->getChar(0) & 0xff) == 0xfe &&
       (s1->getChar(1) & 0xff) == 0xff)
       {
-      isUnicode = gTrue;
+      isUnicode = true;
       i = 2;
       }
     else
       {
-      isUnicode = gFalse;
+      isUnicode = false;
       i = 0;
       }
     while (i < obj.getString()->getLength())
@@ -398,7 +398,7 @@
   std::string creationdate;
   std::string moddate;
 
-  GBool isUnicode = gFalse;
+  bool isUnicode = false;
   if (doc->isOk())
     {
 #ifdef LIBPOPPLER_NEW_OBJECT_API
