Description: Remove vtkgdcm/java and vtkgdcm/c#
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: no
Last-Update: 2022-09-02

--- gdcm-3.0.16.orig/CMakeLists.txt
+++ gdcm-3.0.16/CMakeLists.txt
@@ -706,7 +707,7 @@ if(GDCM_STANDALONE)
         #TEST_OUTPUT_DATA_DIRECTORY "${vtkgdcm_test_data_directory_output}"
       )
 
-      if(GDCM_WRAP_PYTHON AND VTK_WRAP_PYTHON)
+      if(VTKGDCM_WRAP_PYTHON)
         find_package(PythonInterp ${VTK_PYTHON_VERSION} QUIET)
 
         vtk_module_wrap_python(
@@ -739,7 +740,7 @@ if(GDCM_STANDALONE)
           DESTINATION "${GDCM_INSTALL_PACKAGE_DIR}")
       endif()
 
-      if(GDCM_WRAP_JAVA AND VTK_WRAP_JAVA)
+      if(VTKGDCM_WRAP_JAVA)
         # TODO: this is broken, incomplete, needs lots of work
         vtk_module_wrap_java(
           MODULES         ${vtkgdcm_modules}
@@ -901,11 +902,11 @@ if(GDCM_STANDALONE) # disabled for ITK d
   if(GDCM_USE_VTK)
     foreach(comp ${components})
       if( "${comp}" STREQUAL "PythonModule" )
-        if(VTK_WRAP_PYTHON)
+        if(VTK_WRAP_PYTHON AND VTKGDCM_WRAP_PYTHON)
           list(APPEND components VTK${comp})
         endif()
       elseif( "${comp}" STREQUAL "JavaModule" )
-        if(VTK_WRAP_JAVA)
+        if(VTK_WRAP_JAVA AND VTKGDCM_WRAP_JAVA)
           list(APPEND components VTK${comp})
         endif()
       else()
--- gdcm-3.0.16.orig/Utilities/VTK/CMakeLists.txt
+++ gdcm-3.0.16/Utilities/VTK/CMakeLists.txt
@@ -297,7 +297,7 @@ add_custom_target(VTKGDCMPHP ALL
 endif()
 endif()
 
-if(GDCM_WRAP_CSHARP)
+if(VTKGDCM_WRAP_CSHARP)
   if( "${VTK_MAJOR_VERSION}.${VTK_MINOR_VERSION}" LESS 4.5 )
     message(FATAL_ERROR "you need a newer VTK version >= 5.0")
   endif()
